src/main/java/com/seals/camplanner/event/controllers/EventController.java

```diff
@@ -37,10 +47,18 @@ protected Location convert(Long locationId) {
         return toEventDtoList(eventService.findAll());
     }
 
-    //TODO add treatment for the "location" parameter
     @PostMapping("/event")
-    public EventDto saveEvent(@RequestBody EventDto event) {
-        return toDto(eventService.save(fromDto(event)));
+    public EventDto saveEvent(@RequestBody EventDto eventDto) throws LocationUnavaliableException {
+        Event event;
+        try {
+            event = fromDto(eventDto);
+        } catch (MappingException e) {
+            if (e.getCause() instanceof LocationUnavaliableException locationUnavaliableException) {
```

Comment:
SmartCasting, yay!

Reply:
Advantages of using the newst versions of Java hahahaha